tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/HeaderCondition.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public class HeaderCondition
+    {
+        public string Key { get; set; }
+
+        public string ValueRegex { get; set; }
+
+        public string[] MatchedValue { get; set; }
```

Comment:
Feels like a wrong place for this. Imaging a global transform for the entire recording session with requests running in parallel. Are they going to clash?

Reply:
Good point, maybe it should be on RecordEntry.